docs/test-runners/rspec.md

```diff
@@ -0,0 +1,99 @@
+
+### Shift left
+
+First, set up a new test execution job/step/pipeline to run earlier in your software development lifecyle.
+
+Then, to retrieve a subset of tests, first list all the tests you would normally run and pass that to `launchable subset`:
+
+```bash
+launchable subset \
+  --build <BUILD NAME> \
+  --target <TARGET> \
+  rspec spec/**/*_spec.rb > launchable-subset.txt
+```
+
+* The `--build` should use the same `<BUILD NAME>` value that you used before in `launchable record build`.
+* The `--target` option should be a percentage; we suggest `20%` to start. This creates a subset of the most important tests that will run in 20% of the full execution time. As the model learns from your builds, the tests in the subset will become more and more relevant.
+
+This creates a file called `launchable-subset.txt` that you can pass into your command to run tests:
+
+```bash
+bundle exec rspec $(cat launchable-subset.txt) --format d --format RspecJunitFormatter --out rspec.xml
+```
+
+Make sure to continue running the full test suite at some stage. Run `launchable record build` and `launchable record tests` for those runs to continually train the model.
+
+### Shift right
+
+The [shift right](../#shift-right) diagram suggests first splitting your existing test run into two parts:
+
+1. A subset of dynamically selected tests, and
+2. The rest of the tests
+
+To retrieve a subset of tests, first pass the full list of test candidates to `launchable subset`. For example:
+
+```bash
+launchable subset \
+  --build <BUILD NAME> \
+  --target <TARGET> \
+  --rest launchable-remainder.txt \
+  rspec spec/**/*_spec.rb > launchable-subset.txt
+```
+
+* The `--build` should use the same `<BUILD NAME>` value that you used before in `launchable record build`.
+* The `--target` option should be a percentage; we suggest `20%` to start. This creates a subset of the most important tests that will run in 20% of the full execution time. As the model learns from your builds, the tests in the subset will become more and more relevant.
+* The `--rest` option writes all the other tests to a file so you can run them separately.
+
+This creates two files called `launchable-subset.txt` and `launchable-remainder.txt` that you can pass into your command to run tests in two stages:
+
+```bash
+bundle exec rspec $(cat launchable-subset.txt)
+
+bundle exec rspec $(cat launchable-remainder.txt)
+```
+
+You can remove the second part after we've let you know that the model is sufficiently trained. Once you do this, make sure to continue running the full test suite at some stage. Run `launchable record build` and `launchable record tests` for those runs to continually train the model.
+
+## Recording test results
+
+Use [rspec_junit_formatter](https://github.com/sj26/rspec_junit_formatter) to output the test results to a file in RSpec. If you already have a CI server storing your test results it may already be installed.
```

Comment:
@awilkes I think `minitest` also needs `minitest-ci`, but it was not written.  Do you think I should fix that one separately?

Reply:
Yes, sounds good.